integrated_channels/integrated_channel/learner_data.py

```diff
@@ -0,0 +1,215 @@
+"""
+Assist integrated channels with retrieving learner completion data.
+
+Module contains resources for integrated pipelines to retrieve all the
+grade and competion data for enrollments belonging to a particular
+enterprise customer.
+"""
+from __future__ import absolute_import, unicode_literals
+
+from logging import getLogger
+
+from django.utils import timezone
+from django.utils.dateparse import parse_datetime
+from django.utils.translation import ugettext_lazy as _
+from slumber.exceptions import HttpNotFoundError
+
+from enterprise.lms_api import CourseApiClient, GradesApiClient
+from enterprise.models import EnterpriseCourseEnrollment
+from enterprise.utils import NotConnectedToOpenEdX
+
+try:
+    from certificates.api import GeneratedCertificate
+except ImportError:
+    GeneratedCertificate = None
+
+try:
+    from opaque_keys.edx.keys import CourseKey
+except ImportError:
+    CourseKey = None
+
+
+LOGGER = getLogger(__name__)
+
+
+class BaseLearnerExporter(object):
+    """
+    Base class for exporting learner completion data to integrated channels.
+    """
+
+    @property
+    def grade_passing(self):
+        """
+        Returns the string used for a passing grade.
+        """
+        return 'Pass'
+
+    @property
+    def grade_failing(self):
+        """
+        Returns the string used for a failing grade.
+        """
+        return 'Fail'
+
+    @property
+    def grade_incomplete(self):
+        """
+        Returns the string used for an incomplete course grade.
+        """
+        return 'In Progress'
```

Comment:
@e-kolpakov I figured there's a chance that we'd want to send different grade values to different integrated channels, so made them overridable properties instead.